src/fiatconnect/ReviewScreen.tsx

```diff
@@ -34,6 +42,33 @@ export default function FiatConnectReviewScreen({ route, navigation }: Props) {
   const { t } = useTranslation()
   const dispatch = useDispatch()
   const { flow, normalizedQuote, fiatAccount } = route.params
+  const providerId = normalizedQuote.getProviderId()
+
+  const fiatConnectQuotesLoading = useSelector(fiatConnectQuotesLoadingSelector)
+  const fiatConnectQuotes = useSelector(fiatConnectQuotesSelector)
+  const [showingExpiredQuoteDialog, setShowingExpiredQuoteDialog] = useState(false)
+  const [usingUpdatedQuote, setUsingUpdatedQuote] = useState(false)
+
+  let quote = normalizedQuote
+  if (usingUpdatedQuote) {
+    const updatedQuoteData = fiatConnectQuotes.find(
+      (quote) => quote.provider.id === providerId && quote.ok
+    )
+    if (updatedQuoteData) {
+      quote = new FiatConnectQuote({
+        quote: updatedQuoteData as FiatConnectQuoteSuccess,
+        fiatAccountType: normalizedQuote.fiatAccountType,
+        flow,
+      })
+    } else {
+      dispatch(showError(ErrorMessages.QUOTE_UPDATE_FAILED))
```

Comment:
You usually don't want to have a dispatch in the main function path, because this code gets called every time any props, state, or selectors changes values. We'd likely run into bugs with this getting fired multiple times.

Instead, use a `useEffect` hook and have it fire whenever the relevant data (in this case usingUpdatedQuote and quote) changes. Check out [this example in SelectProvider](https://github.com/valora-inc/wallet/blob/main/src/fiatExchanges/SelectProvider.tsx#L101)

Reply:
Gotcha, just updated it!